packages/core/src/queries/user.ts

```diff
@@ -86,12 +85,30 @@ export const hasUserWithIdentity = async (connectorId: string, userId: string) =
 
 export const insertUser = buildInsertInto<CreateUser, User>(pool, Users, { returning: true });
 
-export const findTotalNumberOfUsers = async () => getTotalRowCount(table);
+const buildUserSearchConditionSql = (search: string) => {
+  const searchFields = [fields.primaryEmail, fields.primaryPhone, fields.username, fields.name];
+  const conditions = searchFields.map((filedName) => sql`${filedName} like ${'%' + search + '%'}`);
 
-const findUserMany = buildFindMany<CreateUser, User>(pool, Users);
+  return sql`${sql.join(conditions, sql` or `)}`;
+};
 
-export const findAllUsers = async (limit: number, offset: number) =>
-  findUserMany({ limit, offset });
+export const findTotalNumberOfUsers = async (search: string | undefined) =>
```

Comment:
could update the getTotalRowCount method to take an extra input field for conditional searches? 
i.e. takes a list of key, value set? 

```js
for ({key, value} of conditions) {
  const conditions = searchFields.map((filedName) => sql`${filedName} like ${'%' + value + '%'}`);
  return sql`${sql.join(conditions, sql` or `)}`;
}
```

Reply:
Full conditional support is left to LOG-1599.
Because we not only have `like` operation, we have `>`, `<`.etc.
Also, the `like` has various patterns. not just `%value%`.
So I think it would be better to support this in the future feature.